server/jobmanager/serverv1/jobmanager_server_test.go

```diff
@@ -0,0 +1,349 @@
+
+	assert.Error(t, err)
+}
+
+func Test_jobmanagerServer_Query_JobExists(t *testing.T) {
+	const (
+		owner       = "user1"
+		jobName     = "myJob"
+		programPath = "/bin/ls"
+	)
+	args := []string{"-l", "/"}
+
+	jobManager := jobmanager.NewManagerDetailed(jobmanagertest.NewMockJob, nil)
+	server := serverv1.NewJobManagerServerDetailed(jobManager)
+	ctx := grpcutil.AttachUserIDToContext(context.Background(), owner)
+
+	job, _ := server.Start(ctx, &jobmanagerv1.JobCreationRequest{
+		Name:        jobName,
+		ProgramPath: programPath,
+		Arguments:   args,
+	})
+	jobStatus, err := server.Query(ctx, &jobmanagerv1.JobID{Id: job.Id.Id})
+
+	assert.Nil(t, err)
+	assert.Equal(t, jobName, jobStatus.Job.Name)
+	assert.Equal(t, job.Id, jobStatus.Job.Id)
+	assert.Equal(t, owner, jobStatus.Owner)
+	assert.True(t, jobStatus.IsRunning)
+	assert.Equal(t, int32(1234), jobStatus.Pid)
+	assert.Equal(t, int32(-1), jobStatus.SignalNumber)
+	assert.Equal(t, "", jobStatus.ErrorMessage)
+}
+
+func Test_jobmanagerServer_List_NoUserID(t *testing.T) {
+	jobManager := jobmanager.NewManagerDetailed(jobmanagertest.NewMockJob, nil)
+	server := serverv1.NewJobManagerServerDetailed(jobManager)
+
+	_, err := server.List(context.Background(), &jobmanagerv1.NilMessage{})
+
+	assert.Error(t, err)
+}
+
+func Test_jobmanagerServer_List_NoJobs(t *testing.T) {
+	const owner = "user1"
+	jobManager := jobmanager.NewManagerDetailed(jobmanagertest.NewMockJob, nil)
+	server := serverv1.NewJobManagerServerDetailed(jobManager)
+	ctx := grpcutil.AttachUserIDToContext(context.Background(), owner)
+
+	jobList, err := server.List(ctx, &jobmanagerv1.NilMessage{})
+
+	assert.Nil(t, err)
+	assert.Equal(t, 0, len(jobList.JobStatusList))
+}
+
+func Test_jobmanagerServer_List_JobExists(t *testing.T) {
+	const (
+		owner       = "user1"
+		jobName     = "myJob"
+		programPath = "/bin/ls"
+	)
+	args := []string{"-l", "/"}
+
+	jobManager := jobmanager.NewManagerDetailed(jobmanagertest.NewMockJob, nil)
+	server := serverv1.NewJobManagerServerDetailed(jobManager)
+	ctx := grpcutil.AttachUserIDToContext(context.Background(), owner)
+
+	job, _ := server.Start(ctx, &jobmanagerv1.JobCreationRequest{
+		Name:        jobName,
+		ProgramPath: programPath,
+		Arguments:   args,
+	})
+	jobList, err := server.List(ctx, &jobmanagerv1.NilMessage{})
+
+	assert.Nil(t, err)
+	assert.Equal(t, 1, len(jobList.JobStatusList))
+	assert.Equal(t, jobName, jobList.JobStatusList[0].Job.Name)
+	assert.Equal(t, job.Id, jobList.JobStatusList[0].Job.Id)
+	assert.Equal(t, owner, jobList.JobStatusList[0].Owner)
+	assert.True(t, jobList.JobStatusList[0].IsRunning)
+	assert.Equal(t, int32(1234), jobList.JobStatusList[0].Pid)
+	assert.Equal(t, int32(-1), jobList.JobStatusList[0].SignalNumber)
+	assert.Equal(t, "", jobList.JobStatusList[0].ErrorMessage)
+}
+
+func Test_jobmanagerServer_Stream_NoUserID(t *testing.T) {
+	mockServer := &testserverv1.MockJobmanagerStreamServer{
```

Comment:
As I mentioned in another comment, these are unit tests.  Unit tests should _not_ be integrated with every concrete component they touch.  Such integrations often make unit tests slower, more complex, and more fragile.

Testing things in isolation enables us to have confidence that if one of these test fails, it's because of an error in the code under test (the _unit_ -- the jobmanager sever), not because of an error from some other concrete component with which it interacts.  That isolation then makes it _much_ easier to find and fix errors.

Reply:
Ack.